shell/platform/embedder/embedder.cc

```diff
@@ -166,39 +171,67 @@ FlutterResult FlutterEngineRun(size_t version,
       .gl_make_resource_current_callback = make_resource_current_callback,
   };
 
-  auto platform_view = std::make_shared<shell::PlatformViewEmbedder>(table);
-  platform_view->Attach();
-
-  std::string assets(args->assets_path);
-  std::string main(args->main_path);
-  std::string packages(args->packages_path);
-
-  blink::Threads::UI()->PostTask([
-    weak_engine = platform_view->engine().GetWeakPtr(),  //
-    assets = std::move(assets),                          //
-    main = std::move(main),                              //
-    packages = std::move(packages)                       //
-  ] {
-    if (auto engine = weak_engine) {
-      if (main.empty()) {
-        engine->RunBundle(assets);
-      } else {
-        engine->RunBundleAndSource(assets, main, packages);
-      }
-    }
-  });
-
-  *engine_out = reinterpret_cast<FlutterEngine>(
-      new PlatformViewHolder(std::move(platform_view)));
+  shell::Shell::CreateCallback<shell::PlatformView> on_create_platform_view =
+      [dispatch_table](shell::Shell& shell) {
+        return std::make_unique<shell::PlatformViewEmbedder>(
+            shell,                   // delegate
+            shell.GetTaskRunners(),  // task runners
+            dispatch_table           // embedder dispatch table
+        );
+      };
 
+  shell::Shell::CreateCallback<shell::Rasterizer> on_create_rasterizer =
+      [](shell::Shell& shell) {
+        return std::make_unique<shell::Rasterizer>(shell.GetTaskRunners());
+      };
+
+  // Step 1: Create the engine.
+  auto embedder_engine =
+      std::make_unique<shell::EmbedderEngine>(std::move(thread_host),   //
+                                              std::move(task_runners),  //
+                                              settings,                 //
+                                              on_create_platform_view,  //
+                                              on_create_rasterizer      //
+      );
+
+  if (!embedder_engine->IsValid()) {
+    return kInvalidArguments;
+  }
+
+  // Step 2: Setup the rendering surface.
+  if (!embedder_engine->NotifyCreated()) {
+    return kInvalidArguments;
+  }
+
+  // Step 3: Run the engine.
+  shell::RunConfiguration run_configuration(
+      shell::IsolateConfiguration::CreateForSource(
```

Comment:
Change this to call `RunConfiguration::InferFromSettings`.  `settings` will also need to configure `kernel_snapshot_path` and `application_kernel_path` if a kernel is present in the assets directory.

Reply:
Done.